web-api/src/storage/elastic.rs

```diff
@@ -396,22 +396,51 @@ impl storage::Document for Storage {
         params: KnnSearchParams<'a>,
     ) -> Result<Vec<models::PersonalizedDocument>, Error> {
         // https://www.elastic.co/guide/en/elasticsearch/reference/current/knn-search.html#approximate-knn
+        let excluded_ids = json!({
+            "values": params.excluded.iter().map(AsRef::as_ref).collect_vec()
+        });
+
+        let filter = if let Some(published_after) = params.published_after {
+            // published_after != null && published_after <= publication_date <= now
+            json!({
+                "bool": {
+                    "filter": {
+                        "range": {
+                            "properties.publication_date": {
+                                "gte": published_after.to_rfc3339(),
+                                "lte": "now"
+                            }
+                        }
+                    },
+                    "must_not": {
+                        "ids": excluded_ids
+                    }
+                }
+            })
+        } else {
+            // published_after == null || published_after <= now
+            json!({
+                "bool": {
+                    "must_not": [
+                        { "ids": excluded_ids },
+                        { "range": {
+                            "properties.publication_date": {
+                                "gt": "now"
+                            }
+                        }}
```

Comment:
:art: we should use the same indention as in the other json blocks for readability
```suggestion
                        {
                            "ids": excluded_ids
                        },
                        {
                            "range": {
                                "properties.publication_date": {
                                    "gt": "now"
                                }
                            }
                        }
```